src/controllers/mainController.ts

```diff
@@ -617,6 +617,20 @@ export default class MainController implements vscode.Disposable {
         return false;
     }
 
+    /**
+     * Makes a connection and save if saveConnection is set to true
+     */
+    public async onConnect(uri: string, connectionInfo: IConnectionInfo, connectionPromise: Deferred<boolean>, saveConnection?: boolean): Promise<boolean> {
```

Comment:
docstring for return, please!  Looks like it returns whether the connection is both successful and saved, which wouldn't have been my first guess based on the name of the function